src/compiler/checker.ts

```diff
@@ -11252,6 +11251,27 @@ namespace ts {
                         }
                         return;
                     }
+                    else if (target.flags & TypeFlags.IndexedAccess) {
+                        const targetConstraint = (<IndexedAccessType>target).objectType;
+                        const inference = getInferenceInfoForType(targetConstraint);
+                        if (inference) {
+                            if (!inference.isFixed) {
+                                const map = <MappedType>createObjectType(ObjectFlags.Mapped);
+                                map.templateType = source;
+                                map.constraintType = (<IndexedAccessType>target).indexType;
+                                map.typeParameter = <TypeParameter>createType(TypeFlags.TypeParameter);
+                                // TODO (weswigham): Ensure the name chosen for the unused "K" does not shadow any other type variables in the given scope, so as to not have a chance of breaking declaration emit

```

Comment:
I think the correct way to do this is to set a flag on the symbol which tells the node builder to use an autogenerated identifier for the name, to just ensure there's no conflicts with anything in the surrounding scopes.